src/common/tracer.h

```diff
@@ -67,7 +67,8 @@ struct jspan_context {
   jspan_context(bool sampled_flag, bool is_remote) {}
 };
 
-struct span_stub {
+namespace opentelemetry::trace {
+struct Span {
```

Comment:
because there are places in the code where we use `opentelemetry::trace::Span`

Reply:
where? why are those not guarded by WITH_JAEGER? these wrappers in common/tracer.h are meant to hide/abstract the use of opentelemetry

Reply:
the idea behind creating these dummy classes in `commin/tracer.h` is that the rest of the code is not littered with the `WITH_JAEGER` macro.
especially for the cases where function prototypes depends on that macro.

Reply:
we currently have a `jspan = shared_ptr<opentelemetry::trace::Span>` typedef. if code outside of common/tracer needs to refer directly to the `opentelemetry::trace::Span` type, then we should add a separate typedef for that instead of adding things to another library's namespace. for example, maybe `jspan = opentelemetry::trace::Span` and `jspan_ptr = shared_ptr<Span>`?

Reply:
that said, i couldn't find anything outside of common/tracer that refers to the `Span` type directly. if another PR is adding those, we can move this discussion there

Reply:
I will add the jspan_ptr.
This pr is using the Span typw directly, as it is stored as a bare pointer in the req_context struct

Reply:
oops, no wonder my grep missed it :)

Reply:
fixed here: 0f34499f85ef519f3a64f2ba6efe4755a1098ee6